src/sage/knots/knotinfo.py

```diff
@@ -260,13 +258,15 @@ def eval_knotinfo(string, locals={}, to_tuple=True):
 
     EXAMPLES::
 
-        sage: from sage.knots.knotinfo import KnotInfo, eval_knotinfo
+        sage: from sage.knots.knotinfo import eval_knotinfo
         sage: L = KnotInfo.L4a1_0
         sage: L.braid_notation(original=True)
         '{3, {-2, -2, -1, 2, -1}}'
         sage: eval_knotinfo(_)
         (3, (-2, -2, -1, 2, -1))
     """
+    if not string:
+        raise NotImplementedError('This value is not provided by the database')
```

Comment:
```suggestion
        raise ValueError('this string must not be empty')
```
and good to add a doctest for this.

Reply:
> and good to add a doctest for this.

Of course! Will be done in the next commit.

Reply:
> > and good to add a doctest for this.
> 
> Of course! Will be done in the next commit.

Done in current commit.